tgui/packages/tgui/interfaces/Keybinds/Keybind.tsx

```diff
@@ -19,22 +20,44 @@ interface KeybindProps {
 export const Keybind = (props: KeybindProps) => {
   const { act } = useBackend<KeybindsData>();
   const { keybind, isFocused, setFocusedKey } = props;
+
+  const [lastEvent, setLastEvent] =
+    useState<React.KeyboardEvent<HTMLDivElement>>();
+
   return (
     <LabeledList.Item label={keybind.label}>
       <Button
         onKeyDown={(event) => {
           event.preventDefault();
+          if (!isFocused) {
+            return;
+          }
           if (isEscape(event.key)) {
             setFocusedKey('');
             return;
           }
           if (!isStandardKey(event)) {
+            // Store the event in case the key is released in which case we'll want to bind to this non-Standard key.
+            setLastEvent(event);
             return;
           }
           setFocusedKey('');
           const value = formatKeyboardEvent(event);
           act('changed_key', { id: keybind.id, value });
         }}
+        onKeyUp={(event) => {
+          event.preventDefault();
+          if (!isFocused) {
+            return;
+          }
+          if (!lastEvent) {
+            return;
+          }
+          // If we release a key and we're still in keybind editing mode, it's time to bind whatever we had.
+          setFocusedKey('');
+          const value = formatKeyboardEvent(lastEvent);
```

Comment:
Do we want to clear out the `lastEvent` here as well, or not? Not saying do this, as I haven't tested, just seems odd to hold onto if we're binding it then moving on.